### PR TITLE
file structure, fix type issues

### PR DESCRIPTION
**Pull request by scottrepreneur**
- you don't need to pass the args to the write contract hook call since you already handled them in the prepare config

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -106,7 +105,7 @@ const Home: NextPage = () => {
   };
 
   const handleMintClick = () => {
-    safeMint?.([numTokens]);
```
Comment (scottrepreneur) [inline pages/index.tsx:109]: don't pass args again

**Comment by pcdkd** [comment]
Appreciate you dropping this in. If you have another chance, you think you could take a look at the wagmi/infura/alchemy setup?
Below is correct according to docs, but apiKey returns error that the variable is undefined. 
I have the keys in the .env file but I'm not sure why they're not being passed in here.

```const { chains, provider } = configureChains(
  [polygonMumbai, polygon],
  [
    alchemyProvider({ apiKey: process.env.NEXT_PUBLIC_ALCHEMY_MUMBAI_KEY, priority: 1 }),
    infuraProvider({ apiKey: process.env.NEXT_PUBLIC_INFURA_POLYGON_KEY, priority: 0 }),
    publicProvider({ priority: 2 }),
  ],
  { stallTimeout: 1000 },
);```

**Comment by scottrepreneur** [comment]
It wants you to pass a string. If that env is undefined then it's not happy. You can pass `process.env.ENV_VAR || ""` to secure that type.

**Comment by pcdkd** [comment]
Thanks!

**Pull request closed without merging**